Add root folder for Integration tests/Add sample app for centralized sampling integration tests (#742)
aws-observability/aws-otel-test-framework: centralized-sampling-tests/README.md

# Centralized Sampling Integration Tests

## Introduction

This project is meant to be used to test that X-ray's centralized sampling strategies
are working properly. This folder has sample apps configured for the integration tests
under sample apps in languages that support centralized sampling. As of now the languages
supported are Java and Go. To run these tests, first start up the collector, then
start up the chosen sample app, then start running the tests.

The design doc for this project is found here

> [Design Doc](https://quip-amazon.com/qcsdACbyatNi/X-Ray-Centralized-Sampling-Testing).
> The wiki for this project is found here
> [Wiki](https://w.amazon.com/bin/view/AWS/X-Ray/Development/SDK_and_Tools/Centralized-Sampling-Integration-Testing)

Reviewer: ```suggestion
```

Author: unnecessary links

## Run Locally

### Set up collector
To run locally first set up the collector with the correct configuration.
It is possible that this is already done. The collector-contrib releases can be found here

> [Otel-Collector-Contrib](https://github.com/open-telemetry/opentelemetry-collector-contrib/releases).

Reviewer: ```suggestion
[Otel-Collector-Contrib](https://github.com/open-telemetry/opentelemetry-collector-contrib/releases).
```
Is there a reason this is using upstream collector contrib rather than the ADOT Collector? Also I do not believe this is the correct link to find upstream collector releases.

Author: That was the collector I have been using. No particular reason for using it over the ADOT collector, will change it. They both work the same for this.

Make sure that the collector config is configured to work with a local x-ray listener. 
See the example-collector-config.yaml for what it should look like. Start the collector.

### Start up sample app
Start up the sample app of your choice that is configured for centralized sampling integration
tests. The sample apps exist in the sample-apps folder. Each sample app will have a 
readMe on how to run it. If adding a sample-app to use for the integration tests see

> [Sample-app-requirements](https://quip-amazon.com/o48yA4tsDf4F/Integration-Test-Sample-App-Requirements)

Reviewer: ```suggestion
```

# Centralized Sampling Integration Tests

## Introduction

This project is meant to be used to test that X-ray's centralized sampling strategies
are working properly. This folder has sample apps configured for the integration tests
under sample apps in languages that support centralized sampling. As of now the languages
supported are Java and Go. To run these tests, first start up the collector, then
start up the chosen sample app, then start running the tests.

## Run Locally

### Set up collector
To run locally first set up the collector with the correct configuration.

> It is possible that this is already done. The collector-contrib releases can be found here
> [aws-otel-collector](https://github.com/aws-observability/aws-otel-collector).

Reviewer: ```suggestion
It is possible that this is already done. Available ADOT Collector releases can be found here
[aws-otel-collector](https://github.com/aws-observability/aws-otel-collector/releases).
```

Make sure that the collector config is configured to work with a local x-ray listener pointed
to port 2000. See the example-collector-config.yaml for what it should look like. Start the 
collector with command.
```shell
    docker run --rm -p 2000:2000 -p 55680:55680 -p 8889:8888 \
      -e AWS_REGION=us-west-2 \
      -e AWS_PROFILE=default \
      -v ~/.aws:/root/.aws \

>       -v "${PWD}/examples/docker/config-test.yaml":/otel-local-config.yaml \

Reviewer: Where is this config being copied from?

Author: this was copied from the aws-otel-collector run locally with docker. There will be in later PRs added steps to include github workflows. might make more sense to have run with my example collector though both methods work. The specific config file is under examples/docker/config-test.yaml

      --name awscollector public.ecr.aws/aws-observability/aws-otel-collector:latest \
      --config otel-local-config.yaml;
```

### Start up sample app
Start up the sample app of your choice that is configured for centralized sampling integration

> tests. The sample apps exist in the sample-apps folder. Each sample app will have a 

Reviewer: Which sample-apps in the sample app folder are configured for centralized sampling?

Author: realistically all of the sample apps in this subfolder would be, but that comment was for if other sample apps get added later on when I'm gone for some reason that aren't configured for the testing. Might not be necessary to include that

Reviewer: If you cannot say for sure I would remove this statement. Considering the sample-apps folder hosts sample apps that are used in the collector integration tests. Sample apps for other languages live in their respective language repositories.
